audioipc/src/codec.rs

```diff
@@ -58,9 +58,13 @@ pub struct LengthDelimitedCodec<In, Out> {
 
 enum State {
     Length,
-    Data(u16),
+    Data(usize),
 }
 
+const MAX_MESSAGE_LEN: u64 = 1024 * 1024;
+const MESSAGE_LENGTH_SIZE: usize = std::mem::size_of::<u32>();
+// TODO: static assert that MAX_MESSAGE_LEN can be encoded into MESSAGE_LENGTH_SIZE.
```

Comment:
Is it ok to define the `MAX_MESSAGE_LEN` to its possible max value directly? (`std::u32::MAX`?)

Reply:
I don't think it's useful to support such large message sizes.  A lower limit gives us sensible bounds on the size of allocations.  Most of our IPC messages are tiny (a few tens of bytes), but it's possible for the enumeration of DeviceInfo to be fairly large, so we need to handle "big" messages but there's no need to permit them to be any size at all.